server/routes/profile.js

```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const {
+  profileCreatePost,
+  profileUpdatePost,
+  profileGet,
+  profileGetAll,
+} = require("../controllers/profile");
+router.route("/create").post(protect, profileCreatePost);
+router.route("/update").post(protect, profileUpdatePost);
```

Comment:
For the `update` route, you could make a dynamic route `router.route("/update/:id")` so that `id` is automatically available in the `req.params` object

Reply:
I'm fairly certain that req.user.id is populated with the users data, I used req.user.id as this prevents unauthorized modifications I believe. Let me know what you think about that, thanks!

Reply:
OK, though I can recall you used something like this: `req.params.id || req.user.id`. Which one works for you is fine :smiley: